---
Title: pytestで指定した例外を投げられるかのテストを行うために綺麗な書き方
Category:
- Python
- pytest
Date: 2022-10-11T15:36:18+09:00
URL: https://blog.tricrow.com/entry/python/tips/2022101115
EditURL: https://blog.hatena.ne.jp/good-one/good-one.hatenablog.com/atom/entry/4207112889926488728
---

# 課題

pytestでユニットテストを行うにあたり、例外を投げるタイプの関数をテストしたい。どのようにすればよいか。


# 解決

下記のようにpytest.raisesを使うと綺麗に書ける。[公式参照](https://docs.pytest.org/en/7.1.x/how-to/assert.html)。

    def test_valid():
        with pytest.raises(TypeError, match=r".*max length is 40.*"):
            main.example("1234567890" * 5)
